File: competitive_programming/leetcode/01_python/26_remove_duplicates_sorted_array.py
"""LeetCode 26 - Remove Duplicates from Sorted Array."""
def remove_duplicates(nums):
    result = []
    for value in nums:
        if not result or result[-1] != value:
            result.append(value)
    return result


if __name__ == "__main__":
    assert remove_duplicates([1, 1, 2]) == [1, 2]
    assert remove_duplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4]) == [0, 1, 2, 3, 4]
    print("26 remove duplicates ok")

File: competitive_programming/leetcode/03_cpp/26_remove_duplicates_sorted_array.cpp
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdint>
#include <iostream>
#include <set>
#include <string>
#include <vector>

std::vector<int> removeDuplicates(const std::vector<int> &nums) {
    std::vector<int> result;
    for (int value : nums) if (result.empty() || result.back() != value) result.push_back(value);
    return result;
}

int main() {
    assert((removeDuplicates({1, 1, 2}) == std::vector<int>{1, 2}));
    assert((removeDuplicates({0, 0, 1, 1, 1, 2, 2, 3, 3, 4}) == std::vector<int>{0, 1, 2, 3, 4}));
    std::cout << "26 remove duplicates ok\n";
    return 0;
}
